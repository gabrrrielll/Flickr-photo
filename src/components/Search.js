import React from "react";

const Search = (props) => {
    return (
        <div className="search-block">
            <form onSubmit={props.searchVal}>
                <input id="search" name="search" onChange={props.inputHandler} placeholder="Search here" />
                <span>
                    <button id="submit">Search</button>
                </span>
                <p id="alert">{props.state && props.state.alert !== "" ? props.state.alert : null}</p>
                <p>
                    Your search is: <b> {props.search}</b>
                </p>
            </form>
        </div>
    );
};

export default Search;
