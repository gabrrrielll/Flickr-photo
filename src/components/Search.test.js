import React from "react";
import Search from "./Search";
import { shallow } from "enzyme";

describe("Search component", () => {
    it("should show search button", () => {
        const wrapper = shallow(<Search />);
        const button = wrapper.find("button");
        expect(button).toMatchSnapshot();
    });
});
