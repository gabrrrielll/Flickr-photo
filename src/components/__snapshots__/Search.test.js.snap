// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should show search button 1`] = `ShallowWrapper {}`;
